grease/src/Grease/Setup.hs

```diff
@@ -95,31 +96,40 @@ safeSymbol = W4.safeSymbol . getValueName
 doLog :: MonadIO m => GreaseLogAction -> Diag.Diagnostic -> m ()
 doLog la diag = LJ.writeLog la (SetupDiagnostic diag)
 
-data SetupState sym ext argTys = SetupState
+
+-- | The result of setting up a pointer. `setupResPtr` represents the runtime value of the pointer and
+-- `setupResTgt` stores the runtime representation of what is stored in that pointer. These 
+-- results are memoized during setup based on `BlockId` so that a given block is only given 
+-- one runtime value.
+data SetupRes sym w = SetupRes { setupResPtr :: C.RegValue sym (Mem.LLVMPointerType w), setupResTgt ::  PtrTarget w (C.RegValue' sym)}
```

Comment:
It's probably worth saying *why* we do this memoization in the Haddock for `_setupRes` and referring to that here.
```suggestion
-- | The result of setting up a pointer.
--
-- These results are memoized in the based on the `BlockId` in the `PtrTarget`, see `_setupRes`.
data SetupRes sym w
  = SetupRes
    { -- | The runtime value of the pointer
      setupResPtr :: C.RegValue sym (Mem.LLVMPointerType w)
      -- | The representation of what is stored in that pointer
    , setupResTgt ::  PtrTarget w (C.RegValue' sym)
    }
```

Reply:
(Reading on, I saw that you explained this in the Haddock for `setupPtrMem`, thanks! Probably worth referring to that here)